pkgs/development/tools/parsing/tree-sitter/default.nix

```diff
@@ -71,6 +71,35 @@ let
     in
     lib.mapAttrs change grammars;
 
+  # Usage:
+  # pkgs.tree-sitter.withPlugins (p: [ p.tree-sitter-c p.tree-sitter-java ... ])
+  #
+  # or for all grammars:
+  # pkgs.tree-sitter.withPlugins (_: allGrammars)
+  # which is equivalent to
+  # pkgs.tree-sitter.withPlugins (p: builtins.attrValues p)
+  withPlugins = grammarFn:
+    let
+      grammars = grammarFn builtGrammars;
+    in
+    linkFarm "grammars"
+      (map
+        (drv:
+          let
+            name = lib.strings.getName drv;
+          in
+          {
+            name =
+              (lib.strings.removePrefix "tree-sitter-"
+                (lib.strings.removeSuffix "-grammar" name))
+              + stdenv.hostPlatform.extensions.sharedLibrary;
```

Comment:
isn't this opiniated ?

Reply:
It is, I did it for 2 reasons, because is the format expected by nvim-treesitter (`$LANG_NAME.so`), and because I prefer sorter names. But I'm open to suggestions, I don't mind to change it. But it makes it a bit easier to use the `.so` files with nvim-treesitter (and for now is the only one using the tree-sitter grammars )

Reply:
I also like the shorter names but wanted to check out the reason.

> , because is the format expected by nvim-treesitter ($LANG_NAME.so),

So without the renaming, nvim-treesitter's checkhealth wont work ?

Reply:
Not sure if it won't work, but the `TSInstallInfo` command reports that no grammars are installed (on my current nvim-treesitter, which is 1-2 weeks old)

Reply:
LANG_NAME.so is actually a neovim convention so it's valid concern even without nvim-treesitter and actually justifies the wrapper a bit ^^ Maybe other editors dont follow that convention in which case the renaming function could be an argument but maybe it's too early to add this complexity yet.

Reply:
I agree, it makes sense to wait until other editors adopt tree-sitter